src/app/createlog/page.tsx

```diff
@@ -0,0 +1,232 @@
+import { ExercisesDictionary } from "@/lib/exercises/exercises-dictionary";
+import { useRouter } from "next/navigation";
+import { useEffect, useState } from "react";
+import moduleStyles from "../home-page.module.css";
+
+export default function CreateLog() {
+  const { status, session } = useAuthSession(); // status, session and update are available, see auth-context.tsx
+  console.log(session);
+  const router = useRouter();
+
+  const [searchInput, setSearchInput] = useState("");
+  const [searchResults, setSearchResults] = useState([]);
+  const [selectedExercises, setSelectedExercises] = useState([]);
+
+  const exercisesArray = Object.values(ExercisesDictionary);
+
+  useEffect(() => {
+    if (searchInput) {
+      const results = exercisesArray.filter((exercise) =>
+        exercise.label.toLowerCase().includes(searchInput.toLowerCase())
+      );
+      setSearchResults(results);
+    } else {
+      setSearchResults([]);
+    }
+  }, [searchInput]);
+
+  const handleSelectExercise = (exercise) => {
+    setSelectedExercises((prev) => [
+      ...prev,
+      { ...exercise, sets: [{ reps: 0, weight: 0 }] },
+    ]);
+    setSearchInput(""); // Clear search input after selection
+    setSearchResults([]); // Clear search results after selection
+  };
+
+  const handleAddSet = (index) => {
+    const newSelectedExercises = [...selectedExercises];
+    newSelectedExercises[index].sets.push({ reps: 0, weight: 0 });
+    setSelectedExercises(newSelectedExercises);
+  };
+
+  const handleSetChange = (index, setIndex, field, value) => {
+    const newSelectedExercises = [...selectedExercises];
+    newSelectedExercises[index].sets[setIndex][field] = value;
+    setSelectedExercises(newSelectedExercises);
+  };
+
+  const handleDeleteSet = (index, setIndex) => {
+    const newSelectedExercises = [...selectedExercises];
+    // Remove the set at setIndex from the selected exercise
+    newSelectedExercises[index].sets.splice(setIndex, 1);
+    setSelectedExercises(newSelectedExercises);
+  };
+
+  const handleSaveLog = async () => {
+    // Create sessions array
+    const sessions = [
+      {
+        date: new Date(), // Use current date as an example
+        exercises: selectedExercises.map((exercise) => {
+          // Map selected exercises to exerciseSchema
+          return {
+            exerciseName: exercise.label,
+            sets: exercise.sets.map((set, index) => {
+              // Map sets to setSchema
+              return {
+                setNumber: index + 1, // Set number starts from 1
+                weight: set.weight,
+                unit: "lbs", // Assuming the default unit is pounds
+                reps: set.reps,
+              };
+            }),
+          };
+        }),
+      },
+    ];
+
+    const formData = {
+      userId: session?.user?._id,
+      sessions: sessions,
+    };
+
+    const jsonData = JSON.stringify(formData);
+    console.log(jsonData);
```

Comment:
Typescript is complaining a lot - likely cause there aren't any type annotations. Pls let me know if you need help

Reply:
I'll try fixing this on my end, but will likely need some help. I'll let you know.